### PR TITLE
Report an error if jemalloc.background_thread was requested

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -109,11 +113,36 @@ namespace CurrentMetrics
...
 
 int mainEntryClickHouseServer(int argc, char ** argv)
 {
     DB::Server app;
 
+    auto enable_watchdog = [&]()
+    {
+        if (jemallocOptionEnabled("opt.background_thread"))
+        {
+            LOG_ERROR(&app.logger(), "jemalloc.background_thread has been enabled, but it is not compatible with watchdog (due to fork()), watchdog will not be enabled.");
```
Comment (alexey-milovidov) [inline programs/server/Server.cpp:140]: Not 100% clear, what is the reason?

**Comment by alexey-milovidov** [comment]
For what reasons the user may want to enable this option?
Maybe better to throw exception that this option is unsupported?

We don't encourage users for fine tuning obscure configuration options. If user makes obscure tuning, it's a bad sign.

File: programs/server/Server.cpp (hunk 1 above)
Comment (azat) [reply programs/server/Server.cpp:140]: The reason (in jemalloc) is that background_thread, create threads, and fork() will not clone the threads, and that's why it is disable in jemalloc.

This message should not too verbose I guess, it should just underline the problem.

**Comment by azat** [comment]
> For what reasons the user may want to enable this option?

In general with percpu arena it should not be required, however if percpu_arena was overwritten then background_thread may make sense.

> Maybe better to throw exception that this option is unsupported?

Ok, I'm fine with this approach too.

> We don't encourage users for fine tuning obscure configuration options. If user makes obscure tuning, it's a bad sign.

Agree.

File: programs/server/Server.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply programs/server/Server.cpp]: So, it's an issue with `jemalloc` - it does not work correctly with fork?

**Review by alexey-milovidov — APPROVED** [review]

File: programs/server/Server.cpp (hunk 1 above)
Comment (azat) [reply programs/server/Server.cpp]: Yes (but I guess it can be re-enabled in the child process, manually via mallctl, although not sure that it worth it)

**Pull request merged**